daily-programming/2024-06-26/main0626.cpp

```diff
@@ -0,0 +1,29 @@
+#include <iostream>
+#include <string>
+using namespace std;
+
+int check(string s, int i, int c) {
+  string sub = s.substr(i,1);
+  if(sub == "A" || sub == "C" || sub == "G" || sub == "T") {
+    i++;
+    c++;
+    return check(s, i, c);
+  } else {
+    return c;   
+  }
+}
+
+int main() {
+  string str;
+  cin >> str;
+  int count = 0;
+  int max= 0;
+  for(int i = 0; i < str.size(); i++) {
+    count = check(str, i, count);
+    if(max < count) {
+      max = count;
+    }
+    count = 0;

```

Comment:
多分これでも行ける。というかこっちの方が明確。
```suggestion
    count = check(str, i, 0);
    if(max < count) {
      max = count;
    }
```